### ISSUE TITLE
Boolean flags override even when not specified

### ISSUE THREAD

**Issue by treshugart** [lerna/lerna#895]
When Yargs defaults are specified, they provide the value as flags to the command constructor. This means that when flags, command overrides, lerna json options and legacy options are merged, the flags will take precedence over any thing else because they have a value.

This can easily be reproduced by simply trying to use `stream: true` in the `lerna.json`. If you remove the option from, for example, `RunCommand` it works. Similarly, if you rename it from `boolean` to `string`, it continues to work. So it seems that while `_.defaults()` is doing the right thing (because `false` is a value, where `undefined` is not), providing a default that is considered a "value", will break anything following the durable option convention.

## Expected Behavior

Durable options should take precedence over Yargs defaults.

## Current Behavior

Defaults override user-defined values not specified as flags.

## Possible Solution

After digging a bit deeper, it seems that all we really have to do to get around this is to specify `default: undefined` to any option that is a `boolean` and doesn't already have a default (because they default to `false`. When `_.defaults()` is called, they will then be overridden if they're not specified by the user.

## Steps to Reproduce

1. Put `stream: true` in the `lerna.json`
2. Run a command that should stream.
3. It won't stream.

## Context

- References #893.

**Comment by kring** [comment]
I believe this same issue is preventing `concurrency` from being specified in lerna.json, because it has a default value of 4.

**Comment by treshugart** [comment]
Wrote a test to confirm:

```js
it("should prefer durable options over yargs defaults", () => {
  // yargs does this when default values are specified
  const instance = new TestBCommand([], {
    testOption: false,
    testOption2: 1
  }, testDir);
  expect(instance.options.testOption).toBe("b");
  expect(instance.options.testOption2).toBe("b");
});
```

**Comment by evocateur** [comment]
We could even add an integration test that validates the yargs behaviour. So weird, too, since you'd think an option that _wasn't_ passed would default to `undefined` anyway. (An issue to raise upstream, perhaps?)

**Comment by treshugart** [comment]
It's kinda the whole idea of defaults, though, to provide a value when one isn't specified. If we removed all specified defaults from yargs and had a defaults for each command that we merge in later then that would work, too.

I'm hesitant to raise an issue with yargs as this really feels like the intended behaviour. I think it can probably be worked around it easily if yargs defaults just aren't used. It makes some sense, too, because there are several other places that can specify options within Lerna.

I'm spending some time spiking some solutions. I'll PR what I have when ready and we can go from there, maybe.

**Comment by treshugart** [comment]
Implemented in #898.
